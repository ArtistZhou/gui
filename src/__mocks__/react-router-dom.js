import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
