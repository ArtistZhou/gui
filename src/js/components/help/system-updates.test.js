import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SystemUpdates from './system-updates';
import { helpProps } from './mockData';

describe('SystemUpdates Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SystemUpdates {...helpProps} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
