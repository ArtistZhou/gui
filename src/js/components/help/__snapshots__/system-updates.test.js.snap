// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    System Updates
  </h2>
  <p>
    The built-in update mechanism in Mender is dual A/B rootfs updates, meaning the whole system on the device is updated. To enable system updates, the device needs to have an A/B partition layout with the Mender client installed in its disk image.
  </p>
  <p>
    Learn how to enable and connect your devices for system updates:
  </p>
  <ul>
    <li>
      <p>
        <a
          href="/help/system-updates/board-integrations"
          onClick={[Function]}
        >
          Supported board integrations on Mender Hub
        </a>
      </p>
    </li>
    <li>
      <p>
        <a
          href="/help/system-updates/build-with-yocto"
          onClick={[Function]}
        >
          Building a Mender-enabled Yocto image
        </a>
      </p>
    </li>
    <li>
      <p>
        <a
          href="/help/system-updates/integrate-debian"
          onClick={[Function]}
        >
          Devices running Debian family OSes
        </a>
      </p>
    </li>
  </ul>
  <p>
    You can learn more about Mender's A/B partition architecture
     
    <a
      href="https://docs.mender.io/undefinedarchitecture/overview"
      target="_blank"
    >
      here
    </a>
    .
  </p>
</div>
`;
